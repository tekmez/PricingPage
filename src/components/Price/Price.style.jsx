import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  listContainer: {
    marginTop: 8,
    gap: 12,
  },
  container: {
    paddingVertical: 13,
    paddingHorizontal: 20,
    borderRadius: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  primaryBorder: {
    borderWidth: 1,
    borderColor: '#E7E8E9',
  },
  secondaryBorder: {
    borderWidth: 2,
    borderColor: '#FF0000',
  },
  title: {
    fontSize: 12,
    fontFamily: 'Inter-Regular',
    color: '#717578',
    lineHeight: 16,
    letterSpacing: 0.5,
  },
  priceContainer: {
    flexDirection: 'row',
    gap: 6,
  },
  lightPrice: {
    fontSize: 12,
    color: '#717578',
    fontFamily: 'Inter-Regular',
    lineHeight: 16,
  },
  boldPrice: {
    fontSize: 12,
    fontWeight: '600',
    fontFamily: 'Inter-Bold',
    lineHeight: 16,
  },
})

export default styles
