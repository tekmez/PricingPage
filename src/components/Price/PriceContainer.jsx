import React from 'react'
import { Text, View } from 'react-native'
import PriceBadge from '../PriceBadge'
import styles from './Price.style'

const PriceContainer = ({ saving, title, price, boldPrice }) => {
  return (
    <View>
      {saving && <PriceBadge title={saving} />}
      <Text testID="price-container-title" style={styles.title}>
        {title}
      </Text>
      <View style={styles.priceContainer}>
        {price && (
          <Text testID="light-price-title" style={styles.lightPrice}>
            {price}
          </Text>
        )}
        <Text testID="bold-price-title" style={styles.boldPrice}>
          {boldPrice}
        </Text>
      </View>
    </View>
  )
}

export default PriceContainer
