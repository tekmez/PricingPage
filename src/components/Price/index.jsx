export { default } from './Price'
export { default as PriceList } from './PriceList'
