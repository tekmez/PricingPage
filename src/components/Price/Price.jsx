import React, { useState } from 'react'
import { View } from 'react-native'
import styles from './Price.style'
import RadioButton from '../Radio'
import PriceContainer from './PriceContainer'
const Price = ({ title, saving, price, boldPrice, theme, selected }) => {
  const [isEnabled, setIsEnabled] = useState(selected)
  return (
    <View
      style={[
        styles.container,
        theme === 'primary' ? styles.primaryBorder : styles.secondaryBorder,
      ]}
    >
      <PriceContainer
        saving={saving}
        title={title}
        price={price}
        boldPrice={boldPrice}
      />
      <RadioButton isEnabled={isEnabled} setIsEnabled={setIsEnabled} />
    </View>
  )
}

export default Price
