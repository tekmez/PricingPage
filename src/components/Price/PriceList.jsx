import React from 'react'
import Price from './Price'
import { View } from 'react-native'
import styles from './Price.style'
const PriceList = () => {
  const prices = [
    {
      title: 'YEARLY ACCESS',
      saving: 'SAVE %89',
      price: '$0,11/day',
      boldPrice: '$39.99/year',
      theme: 'primary',
      selected: false,
    },
    {
      title: '30-DAY FREE TRIAL',
      saving: '',
      price: '',
      boldPrice: 'then $6.99/week',
      theme: 'secondary',
      selected: true,
    },
  ]
  return (
    <View style={styles.listContainer}>
      {prices.map((price, index) => (
        <Price key={index} {...price} />
      ))}
    </View>
  )
}

export default PriceList
