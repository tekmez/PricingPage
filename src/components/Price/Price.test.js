import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Price from './Price'
import styles from './Price.style'
jest.mock('../../../assets/svg/checkIcon.svg', () => () => (
  <div testID="check-icon" />
))
describe('Price Component', () => {
  const mockData = {
    title: 'Product Title',
    saving: 'Save 30%',
    price: '19.99',
    boldPrice: '14.99',
    theme: 'primary',
    selected: true,
  }

  it('should match snapshot', () => {
    const comp = render(<Price {...mockData} />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('renders Price component with correct data', () => {
    const comp = render(<Price {...mockData} />)
    const priceContainer = comp.getByTestId('price-container')
    const priceTitle = comp.getByTestId('price-title').children[0]
    const priceText = comp.getByTestId('light-price-title').children[0]
    const boldPriceText = comp.getByTestId('bold-price-title').children[0]
    expect(priceContainer).toBeTruthy()
    expect(priceTitle).toBe(mockData.saving)
    expect(priceText).toBe(mockData.price)
    expect(boldPriceText).toBe(mockData.boldPrice)
  })

  it('should render check icon if selected', () => {
    const comp = render(<Price {...mockData} />)
    const checkIcon = comp.getByTestId('check-icon')
    expect(checkIcon).toBeTruthy()
  })

  it('should correct theme styles', () => {
    const comp = render(<Price {...mockData} />)
    const priceContainer = comp.getByTestId('price-container')
    expect(priceContainer.props.style[1]).toMatchObject(styles.primaryBorder)
  })

  it('should secondary theme styles', () => {
    const comp = render(<Price {...mockData} theme="secondary" />)
    const priceContainer = comp.getByTestId('price-container')
    expect(priceContainer.props.style[1]).toMatchObject(styles.secondaryBorder)
  })
})
