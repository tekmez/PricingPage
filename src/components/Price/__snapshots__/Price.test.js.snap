// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Component should match snapshot 1`] = `
<View
  style={
    [
      {
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 13,
      },
      {
        "borderColor": "#E7E8E9",
        "borderWidth": 1,
      },
    ]
  }
  testID="price-container"
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#FF0000",
          "borderRadius": 16,
          "height": 20,
          "left": 245,
          "paddingHorizontal": 16,
          "paddingVertical": 2,
          "position": "absolute",
          "top": -22,
          "width": 92,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "lineHeight": 16,
          }
        }
        testID="price-title"
      >
        Save 30%
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#717578",
          "fontFamily": "Inter-Regular",
          "fontSize": 12,
          "letterSpacing": 0.5,
          "lineHeight": 16,
        }
      }
      testID="price-container-title"
    >
      Product Title
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 6,
        }
      }
    >
      <Text
        style={
          {
            "color": "#717578",
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "lineHeight": 16,
          }
        }
        testID="light-price-title"
      >
        19.99
      </Text>
      <Text
        style={
          {
            "fontFamily": "Inter-Bold",
            "fontSize": 12,
            "fontWeight": "600",
            "lineHeight": 16,
          }
        }
        testID="bold-price-title"
      >
        14.99
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
    testID="radio-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FF0000",
          "borderColor": "#FF0000",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
      testID="radio-button-style"
    >
      <div
        testID="check-icon"
      />
    </View>
  </View>
</View>
`;
